### PR TITLE
JP-3540: remove migrate_data and move_wcs scripts

### LINKED ISSUE #8284: Remove migrate_data.py and move_wcs.py from repository
_Issue [JP-3540](https://jira.stsci.edu/browse/JP-3540) was created on JIRA by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter):_

The scripts move_wcs.py and migrate_data.py were both used to make pre-flight data compatible with what is now the standard data structure.  These scripts may not be used anymore, which means they should no longer be maintained as part of the pipeline repository.  This came up during handling of JP-2090 but deserves its own ticket.

Please comment:
 * Are these scripts still being used? 
 * What potential reasons are there to keep these in the repository?

### PR DESCRIPTION
**Pull request by emolter**
Resolves [JP-3540](https://jira.stsci.edu/browse/JP-3540)

Closes #8284

This PR addresses removal of migrate_data.py and move_wcs.py scripts, which were used a long time ago to update data to what is now the standard format, but are no longer used and therefore need not be maintained as part of the pipeline.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: jwst/scripts/tests/test_scripts.py [hunk 1]
```diff
@@ -21,9 +21,6 @@
     'jump',
     'linearitycorr',
     'make_header',
-    'migrate_data',
-    'minimum_deps',
```
Comment (hbushouse) [inline jwst/scripts/tests/test_scripts.py:25]: Does anyone know what the entry for `minimum_deps` was for? Never heard of it.

Comment (braingram) [reply jwst/scripts/tests/test_scripts.py:25]: That script (used to test the oldest versions of all dependencies in the `oldestdeps` CI job) was moved to: https://github.com/spacetelescope/minimum_dependencies
Removing it from this test makes sense to me.

Comment (hbushouse) [reply jwst/scripts/tests/test_scripts.py:25]: Thanks @braingram . In that case, I approve of this PR.

**Review by hbushouse — APPROVED** [review]

File: jwst/scripts/tests/test_scripts.py (hunk 1 above)
Comment (emolter) [reply jwst/scripts/tests/test_scripts.py:25]: no idea, but it was making the unit test for the scripts fail locally

**Pull request merged**